nixos/modules/services/system/swapspace.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.swapspace;
+in {
+  ###### interface
+
+  options = {
+
+    services.swapspace = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to create swapfiles dynamically using the SwapSpace manager.
+          Files will be added and removed based on current memory usage.
+        '';
+      };
+
+      path = mkOption {
+        type = types.path;
+        default = "/var/lib/swap";
+        description = ''
+          Location of the swap files. This directory will be restricted to root.
+        '';
+      };
+
+      cooldown = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Cooldown period between changes.
+          SwapSpace will wait at least this many seconds before an action,
+          like removing a swapfile after adding one.
+        '';
+      };
+
+      minSwapSize = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Minimum size of a swapfile.
+        '';
+      };
+
+      maxSwapSize = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Maximum size of a swapfile.
```

Comment:
I think it would be great to explain what happens when this is set to null. Same for minSwapSize.